### PR TITLE
Do keyspace filtering when possible with `show vitess_shards`

### PR DESCRIPTION
**Pull request by aquarapid**
When using a `show vitess_shards like 'keyspace1/%'`
filter, actually use the keyspace part of the query to limit the
keyspaces for which we fetch all shards, instead of getting all
shards from all keyspaces the vtgate knows about and then
filtering them out. This probably does not matter unless you
have a lot of shards and/or are running `show vitess_shards`
like this frequently.

### CODE REVIEW

**Comment by zhongr3n** [comment]
Thanks for quick fix. This looks good!

One minor note: can you add a test case for something like `show vitess_shards like 'keyspace%/%'`. think it should work as expected but would be good to verify

File: go/vt/vtgate/executor_test.go [hunk 1]
```diff
@@ -701,6 +701,36 @@ func TestExecutorShow(t *testing.T) {
...
+	wantqr = &sqltypes.Result{
+		Fields: buildVarCharFields("Shards"),
+		Rows: [][]sqltypes.Value{
+			buildVarCharRow("TestSharded/-20"),
+			buildVarCharRow("TestSharded/e0-"),
+		},
+	}
+	utils.MustMatch(t, wantqr, qr, query)
+
+	query = "show vitess_shards like 'TestSharded%/%'"
```
Comment (deepthi) [inline go/vt/vtgate/executor_test.go:719]: nit: the test will be better if you used 'TestShard%/%' to exercise the regex.

**Review by deepthi — APPROVED** [review]

**Review by systay — APPROVED** [review]

File: go/vt/vtgate/executor_test.go (hunk 1 above)
Comment (zhongr3n) [reply go/vt/vtgate/executor_test.go:719]: +1

**Review by zhongr3n — APPROVED** [review]

**Review by frouioui — APPROVED** [review]

File: go/vt/vtgate/executor_test.go (hunk 1 above)
Comment (aquarapid) [reply go/vt/vtgate/executor_test.go]: Made that change.

**Pull request merged**